### ISSUE TITLE
documentation state.file.managed parameter template not reflecting TEMPLATE_REGISTRY

### ISSUE THREAD

**Issue by Sylvain303** [saltstack/salt#35336]
### Description of Issue/Question

Issue about the documentation:
https://docs.saltstack.com/en/latest/ref/states/all/salt.states.file.html#salt.states.file.managed

it is said:

> **template**
>   If this setting is applied then the named templating engine will be used to render the downloaded file, currently jinja, mako, and wempy are supported

The templating engine limitation is false, and not only limited to  what is written:

* The code of [salt.states.file.managed](https://github.com/saltstack/salt/blob/develop/salt/states/file.py#L1647)
* uses [`modules.file.apply_template_on_content`](https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.file.html#salt.modules.file.apply_template_on_contents) code [here](https://github.com/saltstack/salt/blob/develop/salt/modules/file.py#L3430).
~~~python
        # Apply templating
        contents = salt.utils.templates.TEMPLATE_REGISTRY[template](
            contents,
            from_str=True,
            to_str=True,
            context=context_dict,
            saltenv=saltenv,
            grains=__grains__,
            pillar=__pillar__,
            salt=__salt__,
opts=__opts__)['data'].encode('utf-8')
~~~
* it uses [TEMPLATE_REGISTRY](https://github.com/saltstack/salt/blob/develop/salt/utils/templates.py#L544)

~~~python
TEMPLATE_REGISTRY = {
    'jinja': JINJA,
    'mako': MAKO,
    'py': py,
    'wempy': WEMPY,
    'genshi': GENSHI,
    'cheetah': CHEETAH,
}
~~~

It is confusing and should be changed.
May be the template engines available can be listed in some way.

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)
~~~
salt-call --versions-report
Salt Version:
           Salt: 2016.3.2
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.2
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
         Jinja2: 2.7.3
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.2
   mysql-python: 1.2.3
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.9 (default, Mar  1 2015, 12:57:24)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.4.0
           RAET: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
 
System Versions:
           dist: debian 8.5 
        machine: x86_64
        release: 3.16.0-4-amd64
         system: Linux
        version: debian 8.5 
~~~

## example of template: `py` 
works fine, but hard to guess…

state.sls

~~~yaml
/tmp/pipo:
  file.managed:
    - template: py
    - source: salt://pipo.py
    - defaults:
        molo: some one fall into the watter
~~~

pipo.py

~~~python
#!py
def run():
    molo = globals()['context']['molo']
    return molo + ','.join(globals().keys())
~~~

**Comment by Ch3LL** [comment]
@Sylvain303 thanks for pointing this out. Looks like we need to add more renderers to that list in the documentation.

**Comment by rallytime** [comment]
Thanks @Sylvain303 - I have fixed this with #35360.

**Comment by Sylvain303** [comment]
Your change seems OK. 
cool.  :)
When will it be pushed on line?

Could it be links to related renders, also?

Is there an information command that could list available template engine in our current salt installation already?

**Comment by rallytime** [comment]
> When will it be pushed on line?
Hm. It should already be live on 2016.3. I think something must be up with our doc building. I'll look into that.

> Could it be links to related renders, also?
Definitely! Good idea. I'll make a follow-up PR and link it here.

> Is there an information command that could list available template engine in our current salt installation already?
Not that I know of. Want to file another issue as a feature request for that type of functionality?

**Comment by rallytime** [comment]
I wasn't able to figure out how to fix the doc builds presently. I have filed a new issue about that in #35399.

The link additions to each renderer have been added to the docs in #35406.

**Comment by Sylvain303** [comment]
in #35408 I looked at the output of `salt '*' sys.doc` a lot of references to `template` are listed here.

May be the doc is false at many places in fact…

Still no doc update online, when it is supposed to occur? 
Is it bugged as you suggest in #35399, or is it related to version release process?

**Comment by rallytime** [comment]
@Sylvain303 It is a bug in our doc building process. I have asked @jacobhammons to look into it on the issue that I made.

Also, I meant to look for other template references in this file, but I clearly forgot to follow up on that. Thanks for the reminder.

**Comment by rallytime** [comment]
@Sylvain303 I have amended the other `template` doc references to include the list of available templates in #35498. Thanks for following up here.
